src/main/java/com/stosik/parking/reservation/domain/ReservationFacade.java

```diff
@@ -0,0 +1,101 @@
+package com.stosik.parking.reservation.domain;
+
+import com.stosik.parking.reservation.domain.evaluator.PriceCalculator;
+import com.stosik.parking.reservation.domain.model.Reservation;
+import com.stosik.parking.reservation.dto.CreateReservationCommand;
+import com.stosik.parking.reservation.dto.ReservationDto;
+import com.stosik.parking.reservation.exceptions.CreateReservationException;
+import com.stosik.parking.reservation.exceptions.ReservationNotFoundException;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.Objects;
+import java.util.Optional;
+
+@Transactional
+@Slf4j
+public class ReservationFacade
+{
+    private final ReservationRepository reservationRepository;
+    private final PriceCalculator priceCalculator;
+    private final Meter parkingMeter;
+    private final ParkingStore parkingStore;
+    private final ReservationDtoCreator reservationDtoCreator;
+    
+    ReservationFacade(ReservationRepository reservationRepository, ParkingStore parkingStore, PriceCalculator priceCalculator,
+        Meter parkingMeter, ReservationDtoCreator reservationDtoCreator)
+    {
+        this.reservationRepository = reservationRepository;
+        this.parkingStore = parkingStore;
+        this.priceCalculator = priceCalculator;
+        this.parkingMeter = parkingMeter;
+        this.reservationDtoCreator = reservationDtoCreator;
+    }
+    
+    public ReservationDto startParkmeter(CreateReservationCommand createReservationCommand)
+    {
+        return Optional
+            .ofNullable(createReservationCommand)
```

Comment:
kiedy `createReservationCommand` byłby nullem?

Reply:
Faktycznie, w tym przypadku nie ma to uzasadnienia, ponieważ utworzenie rezerwacji w przypadku nie startowania parametru nie występuje, natomiast nigdy w systemie nie spotkamy się z wartością null.